### PR TITLE
expose a few zarr backend functions as semi-public api

### LINKED ISSUE #3851: Exposing Zarr backend internals as semi-public API
We recently built a prototype REST API for serving xarray datasets via a Fast-API application (see #3850  for more details). In the process of doing this, we needed to use [a few internal functions in Xarray's Zarr backend](https://github.com/jhamman/xpublish/blob/aff49ec09136a29b56167a1d627fcb3a13fa4d01/xpublish/rest.py#L13-L20):

```python
from xarray.backends.zarr import (
    _DIMENSION_KEY,
    _encode_zarr_attr_value,
    _extract_zarr_variable_encoding,
    encode_zarr_variable,
)
from xarray.core.pycompat import dask_array_type
from xarray.util.print_versions import get_sys_info, netcdf_and_hdf5_versions
```

Obviously, none of these imports are really meant for use outside of Xarray's backends so I'd like to discuss how we may go about exposing these functions (or variables) as semi-public (advanced use) API features. Thoughts?

cc @rabernat

### PR DESCRIPTION
**Pull request by jhamman**
- [x] Fixes #3851
 - [x] Tests added
 - [x] Passes `isort -rc . && black . && mypy . && flake8`
 - [ ] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

### CODE REVIEW

File: xarray/backends/zarr.py [hunk 1]
```diff
@@ -10,13 +10,20 @@
 from .common import AbstractWritableDataStore, BackendArray, _encode_variable_name
 
 # need some special secret attributes to tell us the dimensions
-_DIMENSION_KEY = "_ARRAY_DIMENSIONS"
+DIMENSION_KEY = "_ARRAY_DIMENSIONS"
```
Comment (max-sixty) [inline xarray/backends/zarr.py]: We've successfully used `...` to represent all dimensions in `transpose` & `stack`, to the extent we could replicate that here (maybe another time though)

Comment (jhamman) [reply xarray/backends/zarr.py]: This string is used as a special key in Zarr's array metadata so I think this are mostly unrelated to the selection of specific (all) dimensions.

Comment (max-sixty) [reply xarray/backends/zarr.py]: Ah I see, cheers

**Comment by dcherian** [comment]
Add to api.rst?

**Comment by jhamman** [comment]
> Add to api.rst?

Personally, I'm not sure this is required for these functions. For my particular applications, just having some basic tests and doc strings is enough.

**Comment by dcherian** [comment]
Yes I'm not sure either. Just wanted to bring it up. Perhaps  we should add it once the backends third-party interface is ready.

**Pull request merged**